security-admin/src/main/java/org/apache/ranger/rest/XUserREST.java

```diff
@@ -863,6 +868,49 @@ public void deleteXUserByUserName(@PathParam("userName") String userName,
 		xUserMgr.deleteXUser(vxUser.getId(), forceDelete);
 	}
 
+
+	/**
+	 * Proceed with caution: Force deletes users in bulk from the ranger db,
+	 * essentially serves as a cleanup Op for external users.
+	 * <tt>Delete</tt> happens one at a time with immediate commit on the transaction.
+	 */
+	@DELETE
+	@Path("/delete/external/users")
+	@PreAuthorize("hasRole('ROLE_SYS_ADMIN')")
+	@Produces({ "application/json" })
+	public Response deleteXUsers() {
+		if (!usersDeletionInProgress){
+			synchronized (XUserREST.class) {
+				usersDeletionInProgress = true;
+				xUserMgr.bulkDeleteUsers();
+				usersDeletionInProgress = false;
+			}
+			return Response.ok("External users deleted successfully").build();
+		}
+		return Response.ok("Users Deletion in progress, check ranger admin logs!").build();
+	}
+
+	/**
+	 * Proceed with caution: Force deletes groups in bulk from the ranger db,
+	 * essentially serves as a cleanup Op for external groups.
+	 * <tt>Delete</tt> happens one at a time with immediate commit on the transaction.
+	 */
+	@DELETE
+	@Path("/delete/external/groups")
+	@PreAuthorize("hasRole('ROLE_SYS_ADMIN')")
+	@Produces({ "application/json" })
+	public Response deleteXGroups() {
```

Comment:
Instead of an API specifically to delete all external users, consider a generic API that let's the call specify the users to delete via query-params - similar to in searchXUsers():

```
@DELETE
@Path("/groups")
@PreAuthorize("hasRole('ROLE_SYS_ADMIN')")
@Produces({ "application/json" })
public Response deleteGroups(@Context HttpServletRequest request) {
  SearchCriteria searchCriteria = searchUtil.extractCommonCriterias(request, xGroupUserService.sortFields);

  // TODO: update searchCriteria with query-params 

  Long<Long> groupIds = xUserMgr.deleteGroups(searchCriteria);

  return Response.ok(groupIds.size() + " external groups deleted successfully").build();
}
```